geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaDataStore.scala

```diff
@@ -69,6 +70,14 @@ class KafkaDataStore(override val zookeepers: String,
     super[ContentDataStore].dispose()
     super[KafkaDataStoreSchemaManager].dispose()
   }
+
+  def getKafkaSchema(typeName: String): SimpleFeatureType = getSchema(new NameImpl(typeName))
```

Comment:
Looking in ContentDataStore, it looks like you don't want this method or to override getSchema. Instead, you want to modify the schema coming back from the feature source in `createFeatureSource`, which in this case would mean changing the cache-loader on line 48.